firefox-ios/Storage/RustKeychain.swift

```diff
@@ -0,0 +1,68 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+
+open class RustKeychain {
+    public private(set) var serviceName: String
+    public private(set) var accessGroup: String?
+
+    static var sharedClientAppContainerKeychain: RustKeychain {
+        let baseBundleIdentifier = AppInfo.baseBundleIdentifier
+
+        guard let accessGroupPrefix = Bundle.main.object(forInfoDictionaryKey: "MozDevelopmentTeam") as? String else {
+            return RustKeychain(serviceName: baseBundleIdentifier)
+        }
+        return RustKeychain(serviceName: baseBundleIdentifier,
+                            accessGroup: AppInfo.keychainAccessGroupWithPrefix(accessGroupPrefix))
+    }
+
+    public init(serviceName: String, accessGroup: String? = nil) {
+        self.serviceName = serviceName
+        self.accessGroup = accessGroup
+    }
+
+    struct KeychainError: Error {
+        let errorMessage: String
+    }
+
+    func getBaseKeychainQuery(key: String) -> [String: Any?] {
+        let encodedIdentifier: Data? = key.data(using: String.Encoding.utf8)
+        let keychainQueryDictionary: [String: Any?] = [kSecClass as String: kSecClassGenericPassword,
+                                                       kSecAttrAccessGroup as String: self.accessGroup,
+                                                       kSecAttrService as String: self.serviceName,
+                                                       kSecAttrGeneric as String: encodedIdentifier,
+                                                       kSecAttrAccount as String: encodedIdentifier,
+                                                       kSecAttrSynchronizable as String: false]
+        return keychainQueryDictionary
+    }
+
+    func queryKeychainForKey(key: String) -> Result<Data?, Error> {
```

Comment:
Derived from [this data function](https://github.com/mozilla/application-services/blob/c23b4bbc5a225af2a20457178ed5bf0d27927825/components/fxa-client/ios/FxAClient/MZKeychain/KeychainWrapper.swift#L213)